core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -257,10 +262,16 @@ public void startUp(Session session, String keyspace, boolean resetDb, boolean c
                 .put(STRING, Functions::getStringDataPoint)
                 .build();
 
+        metricsIdxCache = CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.HOURS).maximumSize(40000).build();
```

Comment:
@burmanm, it would have made sense to use Infinispan if it was used somewhere else in the code. But to introduce the Infinispan dependency just for this is a bit too much. I will open a ticket for the future to reconsider this decision once we add Infinispan dependency to the project for other reasons.